src/preamble.js

```diff
@@ -742,6 +742,14 @@ function lengthBytesUTF32(str) {
   return len;
 }
 
+// Allocate heap space for a JS string, and write it there.
+function allocateUTF8(str) {
+  var size = lengthBytesUTF8(str) + 1;
+  var ret = _malloc(size);
+  stringToUTF8Array(str, HEAP8, ret, size);
```

Comment:
This would be good to be `if (ret) stringToUTF8Array(str, HEAP8, ret, size);` so that if `malloc` fails, it this will then also gracefully fail and return 0. Also might be good for the comment to explicitly mention that it's caller's responsibility to call `_free()` to free up the memory.

Reply:
Thanks, I did both those things.